src/hooks/useRunChecks.test.ts

```diff
@@ -0,0 +1,56 @@
+import { createK6Check } from '@/test/factories/k6Check'
+import { useRunChecks } from './useRunChecks'
+import { K6Check } from '@/types'
+import { renderHook } from '@testing-library/react'
+import { act } from 'react'
+import { beforeAll, beforeEach, describe, expect, it, vi } from 'vitest'
+
+const onScriptCheck = vi.fn()
+
+beforeAll(() => {
+  window.studio = {
+    ...window.studio,
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    script: {
+      onScriptCheck,
+    },
+  }
```

Comment:
Slightly shorter and won't require eslint/ts comments
```suggestion
  vi.stubGlobal('studio', {
    script: {
      onScriptCheck,
    },
  })
```